realm/realm-library/src/objectServer/java/io/realm/mongodb/sync/SyncConfiguration.java

```diff
@@ -592,8 +581,21 @@ private void validateAndSet(URL baseUrl ) {
             } catch (URISyntaxException e) {
                 throw new IllegalArgumentException("Invalid URI: " + baseUrl, e);
             }
+        }
+
+        /**
+         * FIXME: Make public once https://github.com/realm/realm-object-store/pull/1049 is merged.
+         *
+         * Sets the filename for the Realm file on this device.
+         */
+        Builder name(String filename) {
```

Comment:
It was left out because I wasn't clear exactly how we could support it with the current OS API's. But it is pretty useful, which is why I'm adding it back in the OS PR.

The directory is kinda supported by using `AppConfig.syncRootDirectory()`, which is good enough IMO. I think it make sense for us to control the intermediate layout in order to be able to support multiple Realms from different apps/users.